ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/qos.schema.yml

```diff
@@ -31,3 +31,20 @@ keys:
                 Example: "1 to dscp 32"
       rewrite_dscp:
         type: bool
+      random_detect:
+        type: dict
+        description: |
+          Global random-detect settings
+        keys:
+          ecn:
+            type: dict
+            description:
+              Global ECN Configuration
+            keys:
+              allow:
+                type: str
+                valid_values: ["non-ect", "non-ect chip-based"]
```

Comment:
It seems wrong to have a str value that always have to contain "non-ect".
Maybe change to dict with booleans if there are no other values?
```yaml
keys:
  allow_non_ect:
    type: dict
      keys:
        enabled:
          type: <bool>
        chip_based:
          type: <bool>
```

Reply:
Discussed in chat and agreed it is best to use the dictionary structure.